platform-dom/src/main/java/com/softicar/platform/dom/elements/popup/compositor/DomPopupHierarchyTree.java

```diff
@@ -67,13 +42,27 @@ public List<DomPopup> getAllChildPopups(DomPopup parent) {
 		return childPopups;
 	}
 
+	@Override
+	public Optional<DomPopup> getParentPopup(DomPopup child) {
+
+		Objects.requireNonNull(child);
+		return Optional.ofNullable(parentMap.get(child));
+	}
+
 	/**
-	 * Removes all {@link DomPopup} elements from this
-	 * {@link DomPopupHierarchyTree}.
+	 * Adds the given parent-child relation to this tree.
+	 *
+	 * @param parent
+	 *            the parent {@link DomPopup} (never <i>null</i>)
+	 * @param child
+	 *            the child {@link DomPopup} (never <i>null</i>)
+	 * @throws IllegalArgumentException
+	 *             if the given parent and child references are the same
```

Comment:
That's a lie! At least, I cannot see, where this is thrown. ;)

Reply:
```
private static class ChildMap extends HashMap<DomPopup, List<DomPopup>> {

public void put(DomPopup parent, DomPopup child) {

	Objects.requireNonNull(parent);
	Objects.requireNonNull(child);
	if (parent == child) {
		throw new IllegalArgumentException();           <---------- here
	}
	computeIfAbsent(parent, dummy -> new ArrayList<>()).add(child);
}
}
```